code/modules/surgery/organs/augments_internal.dm

```diff
@@ -253,6 +253,73 @@
 				if(H.stat == CONSCIOUS)
 					H << "<span class='notice'>You feel your heart beating again!</span>"
 
+/obj/item/organ/internal/cyberimp/chest/arm_mod//dummy parent item for making arm-mod implants. works best with nodrop items that are sent to nullspace upon being dropped.
+	name = "Arm-mounted item implant"
+	desc = "You shouldn't see this! Adminhelp and report this as an issue on github!"
+	icon_state = "chest_implant"
+	implant_color = "#007ACC"
+	slot = "shoulders"
+	origin_tech = "materials=5;biotech=4;powerstorage=4"
+	organ_action_name = "Toggle Arm Mod"
+	var/obj/holder//is defined as the retractable item itself. ensure this is defined somewhere!
+	var/out = 0//determines if the item is in the owner's hand or not
+	var/overloaded = 0//is set to 1 when owner gets EMPed. if set to 1, implant doesn't work.
+
+/obj/item/organ/internal/cyberimp/chest/arm_mod/ui_action_click()
+	if(overloaded)//ensure the implant isn't broken
+		owner.visible_message("<span class='warning'>The implant doesn't respond. It seems to be broken...</span>")
+		return
+	if(out)//check if the owner has the item out already
+		owner.unEquip(holder, 1)//if he does, take it away. then,
+		holder.loc = null//stash it in nullspace
+		out = 0//and set this to clarify the item isn't out.
+		owner.visible_message("<span class='notice'>You retract [holder].</span>")
+		playsound(get_turf(owner), 'sound/mecha/mechmove03.ogg', 50, 1)
+	else//if he doesn't have the item out
+		if(owner.put_in_hands(holder))//put it in his hands.
+			out = 1
+			owner.visible_message("<span class='notice'>You extend [holder]!</span>")
+			playsound(get_turf(owner), 'sound/mecha/mechmove03.ogg', 50, 1)
+		else//if this fails to put the item in his hands,
+			holder.loc = null//keep it in nullspace
```

Comment:
Yes, during testing the check would fail and it would place the NODROP item on the ground, which is a big no-no